### ISSUE TITLE
Wrong scope of "Auto enclose expression after superscript or subscript"

### ISSUE THREAD

**Issue by Elaws** [joeyuping/quick_latex_obsidian#34]
The scope of "Auto enclose expression after superscript or subscript" extends beyond the math expression that includes it, which leads to unwanted behavior.

# How to reproduce :

Simply write the following : 

```
$text^3$ some other text. $Another
```

As soon as you will press "space" after the word `Another`, you will get the following result :

```
$text^{3$ some other text. $Another}
```

Notice the unwanted brackets that have been added.

# Workaround :

For now, impossible to use simple expressions such as `^3` without pressing `space` just after to generate `^{3}`.

**Comment by Paval-Andrei** [comment]
@Elaws Looks like I can't reproduce the bug without the closing ```$``` after ```Another```:
``` 
$text^3$ some other text. $Another$
```

But it does produce the result you describe. Anyway, it's also a problem if you like to have a space after ```^```, which I think looks better:
```
\sum_{k = 1} ^ {n} =
```
In this case, if I type ```space``` after ```=```, I get:
```
\sum_{k = 1} ^{ {n} =}
```
It seems to insert the opening bracket ```{``` immediately after ```^``` and the closing bracket  ```}``` immediately before the inserted space, no matter how many characters are between them or if there is another ```^```.

Regarding the workarounds:
1. You can turn off the "Auto enclose expression" features.
2. You can make sure there are brackets ```{}``` after ```^```, with no spaces between them.
3. Looks like I can avoid the bug if, instead of ```space```, I type ```shift + space```.

**Comment by joeyuping** [comment]
Resolved in v2.3.0, please update and take a look!

**Comment by joeyuping** [comment]
@Paval-Andrei I have adjusted the code to allow spacing between the ^ symbol  and the numbers.

e.g.  x ^ {3}   this is possible now. just type x ^ 3  then space bar, the brackets will be added in automatically to become x ^ {3}
